Content.Server/Administration/BwoinkSystem.cs

```diff
@@ -60,15 +97,56 @@ protected override void OnBwoinkTextMessage(BwoinkTextMessage message, EntitySes
             foreach (var channel in targets)
                 RaiseNetworkEvent(msg, channel);
 
+            var sendsWebhook = _webhookUrl != string.Empty;
+            if (sendsWebhook)
+            {
+                var sawmill = IoCManager.Resolve<ILogManager>().GetSawmill("AHELP");
+
+                void LookupFinished(Task<LocatedPlayerData?> finishedTask)
+                {
+                    if (finishedTask.Result == null)
+                    {
+                        sawmill.Log(LogLevel.Error, $"Unable to find player for netuserid {msg.ChannelId} when sending discord webhook.");
+                        return;
+                    }
+
+                    var payload = new WebhookPayload()
+                    {
+                        Username = _serverName,
+                        Content = $"`[{finishedTask.Result.Username}]` {senderSession.Name}: \"{escapedText}\""
+                    };
+                    var request = _httpClient.PostAsync(_webhookUrl,
+                        new StringContent(JsonConvert.SerializeObject(payload), Encoding.UTF8, "application/json"));
+
+                    request.ContinueWith(task =>
+                    {
+                        if (!task.Result.IsSuccessStatusCode)
+                        {
+                            sawmill.Log(LogLevel.Error, $"Discord returned bad status code: {task.Result.StatusCode}");
+                        }
+                    });
+                }
+
+                _playerLocator.LookupIdAsync(msg.ChannelId).ContinueWith(LookupFinished);
+            }
```

Comment:
Please use a proper async method for this. This is very hard to read and reason about, and especially dangerous because it's running from another thread.

Reply:
Please **also** make the method calling `LookupIdAsync` async so I don't have to look at this `ContinueWith` stuff.